src/Skeleton/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { FunctionComponent } from 'react'
+
+import clsx from 'clsx'
+import styles from './skeleton.module.scss'
+
+interface Props {
+  /** Sets the skeleton-pulse style */
+  type?: 'text' | 'image' | 'button' | 'button-secondary'
+  /** Sets the width of the skeleton. */
+  width?: number | string | []
+  /** Sets the height of the skeleton. */
+  height?: number | string | []
+  /** Makes the skeleton look like a circle */
+  circle?: boolean
+  [rest: string]: unknown // ...rest property
+}
+
+const Skeleton: FunctionComponent<Props> = ({ type = 'text', width, height, circle, ...rest }) => {
```

Comment:
just define style and className separately from rest and we don't want to spread those to the span as it'd override the styles and className.

Reply:
Added both style and className props.. can you check this?